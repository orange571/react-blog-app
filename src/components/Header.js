import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin, startLogout } from '../actions/auth';

export const Header = ({ startLogout, startLogin, displayName }) => (
  <header className="header">
    <div className="show-for-desktop content-container content-container--header">
      <div className="header__content">
          {
            displayName ? (
              <div  className="header__left-nav">
                <Link className="header__title" to="/dashboard">
                  <h1>Thought Collector</h1>
                </Link>
                <NavLink to="/dashboard" className="header__link" activeClassName="is-active" exact={true}>Dashboard</NavLink>
                <NavLink to="/public" className="header__link "activeClassName="is-active">Browse</NavLink>
              </div>
            ) : (
              <div  className="header__left-nav">
                <Link className="header__title" to="/dashboard">
                  <h1>Thought Collector</h1>
                </Link>
                <Link className="header__link" to="/public">
                  Browse
                </Link>
              </div>
            )
          }
        {
          displayName ? (
            <div className="header__right-nav">
              <div className="header__username">Signed in as {displayName}</div>
              <button className="button button--link logout" onClick={startLogout}>Logout</button>
            </div>
          ) : (
            <div className="header__right-nav">
              <button className="button login" onClick={startLogin}>Login</button>
            </div>
          )
        }
      </div>
    </div>
    <div className="show-for-mobile">
      <div  className="header_content--top">
        <Link className="header__title" to="/dashboard">
          <h1>Thought Collector</h1>
        </Link>
        {displayName && <div className="header__username">Signed in as {displayName}</div>}
      </div>
      {
        displayName ? (
          <div className="header_content--bottom">
            <div className="content-container--bottom-nav">
              <NavLink to="/dashboard" className="header__link" activeClassName="is-active" exact={true}>Dashboard</NavLink>
              <NavLink to="/public" className="header__link "activeClassName="is-active">Browse</NavLink>
              <button className="button button--link logout" onClick={startLogout}>Logout</button>
            </div>
          </div>
        ) : (
          <div className="header_content--bottom">
            <div className="content-container--bottom-nav">
              <Link className="header__link" to="/public">Browse</Link>
              <button className="button button--link login" onClick={startLogin}>Login</button>
            </div>
          </div>
        )
      }
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
  startLogin: () => dispatch(startLogin()),
});


const mapStateToProps = (state) => {
  return {
    displayName: state.auth.displayName
  };
};


export default connect(mapStateToProps, mapDispatchToProps)(Header);
