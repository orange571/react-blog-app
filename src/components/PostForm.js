import React from 'react';
import moment from 'moment';

export default class PostForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      title: props.post ? props.post.title : '',
      body: props.post ? props.post.body : '',
      createdAt: props.post ? moment(props.post.createdAt) : moment(),
      author: props.post ? props.post.author: props.displayName,
      error: ''
    };
  }
  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };
  onBodyChange = (e) => {
    const body = e.target.value;
    this.setState(() => ({ body }));
  };
  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.title || !this.state.body) {
      this.setState(() => ({ error: 'Please provide a title and post body.' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        title: this.state.title,
        createdAt: this.state.createdAt.valueOf(),
        body: this.state.body,
        author: this.state.author.valueOf(),
      });
    }
  };
  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <input
          type="text"
          placeholder="Title"
          autoFocus
          className="text-input"
          value={this.state.title}
          onChange={this.onTitleChange}
        />
        <textarea
          placeholder="Add a body for your post"
          className="textarea"
          value={this.state.body}
          onChange={this.onBodyChange}
        >
        </textarea>
        <div className="form__button-container">
          <button type="submit" className="button">Save Post</button>
          {this.props.onRemove && (<button className="button button--secondary" onClick={this.props.onRemove}>Remove Post</button>)}
        </div>
      </form>
    )
  }
}
