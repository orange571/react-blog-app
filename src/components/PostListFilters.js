import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { setTextFilter, sortByDate, sortByTitle} from '../actions/filters';

export class PostListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'title') {
      this.props.sortByTitle();
    }
  };
  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__left">
            <div className="input-group__item">
              <input
                type="text"
                className="text-input"
                placeholder="Search posts"
                value={this.props.filters.text}
                onChange={this.onTextChange}
              />
            </div>
            <div className="input-group__item">
              <select
                className="select"
                value={this.props.filters.sortBy}
                onChange={this.onSortChange}
              >
                <option value="title">Title</option>
                <option value="date">By Date</option>
              </select>
            </div>
          </div>  
          <div className="input-group__item">
                <Link className="button" to="/create">Add Post</Link>
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByTitle: () => dispatch(sortByTitle()),
});

export default connect(mapStateToProps, mapDispatchToProps)(PostListFilters);
