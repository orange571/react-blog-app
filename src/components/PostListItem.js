import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const PostListItem = ({ id, title, body, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <h3 className="list-item__title list-item__title--private">{title}</h3>
    <h3 className="list-item__data">{moment(createdAt).format('M/D/YY')}</h3>
  </Link>
);

export default PostListItem;
