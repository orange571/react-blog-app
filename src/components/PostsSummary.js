/**

import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
//import numeral from 'numeral';
//import selectPosts from '../selectors/posts';

export const PostsSummary = (/**{postCount, postsTotal }**/) => {
  //const postWord = postCount === 1 ? 'post' : 'posts';
  //const formattedPostsTotal = numeral(postsTotal / 100).format('$0,0.00');
/**
  return (
    <div className="page-header">
      <div className="content-container">
        //<h1 className="page-header__title">Viewing <span>{postCount}</span> {postWord} totalling <span>{formattedPostsTotal}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Post</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  //const visiblePosts = selectPosts(state.posts, state.filters);

  return {
    postCount: visiblePosts.length,
    postsTotal: selectPostsTotal(visiblePosts)
  };
};

export default connect(mapStateToProps)(PostsSummary);
**/