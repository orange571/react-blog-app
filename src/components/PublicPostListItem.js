import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const PublicPostListItem = ({ id, title, body, createdAt }) => (
  <Link className="list-item list-item--public" to={`/view/${id}`}>
    <div>
      <h3 className="list-item__title">{title}</h3>
      <div className="list-item__body">{body}</div>
    </div>  
  </Link>
);

export default PublicPostListItem;
