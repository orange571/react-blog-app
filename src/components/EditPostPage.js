import React from 'react';
import { connect } from 'react-redux';
import PostForm from './PostForm';
import { startEditPost, startRemovePost } from '../actions/posts';

export class EditPostPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isReadable : props.post ? props.post.isReadable : false,
    };
  }
  onSubmit = (post) => {
    this.props.startEditPost(this.props.post.id, post);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemovePost({ id: this.props.post.id });
    this.props.history.push('/');
  };
  displayLink = (isReadable) => {
    this.setState(()=>({isReadable:isReadable}));
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Post</h1>
            {this.state.isReadable ? (
                <div id="link-container">
                  <div>Use the link below to share with others</div>
                  <a href={"/view/"+ this.props.match.params.id} target="_blank">https://thought-collector-app.herokuapp.com/view/{this.props.match.params.id}</a>
                </div>
              ) : (
                <div id="link-container">
                  <div>Get a link to share your post with others</div>
                  <div>when you check "Make Public"</div>
                </div>
              )
            }
          </div>
        </div>
        <div className="content-container">
          <PostForm
            post={this.props.post}
            onSubmit={this.onSubmit}
            onRemove={this.onRemove}
            displayLink={this.displayLink}
          />
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  post: state.posts.find((post) => post.id === props.match.params.id),
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditPost: (id, post) => dispatch(startEditPost(id, post)),
  startRemovePost: (data) => dispatch(startRemovePost(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditPostPage);
