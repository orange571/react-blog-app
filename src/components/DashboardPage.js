import React from 'react';
import PostList from './PostList';
import PostListFilters from './PostListFilters';
//import PostsSummary from './PostsSummary';

const DashboardPage = () => (
  <div>
    <PostListFilters />
    <PostList />    
  </div>
);

export default DashboardPage;
