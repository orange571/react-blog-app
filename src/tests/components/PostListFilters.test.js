import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { PostListFilters } from '../../components/PostListFilters';
import { filters, altFilters } from '../fixtures/filters';

let setTextFilter, sortByDate, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  sortByDate = jest.fn();
  wrapper = shallow(
    <PostListFilters
      filters={filters}
      setTextFilter={setTextFilter}
      sortByDate={sortByDate}
    />
  );
});

test('should render PostListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render PostListFilters with alt data correctly', () => {
  wrapper.setProps({
    filters: altFilters
  });
  expect(wrapper).toMatchSnapshot();
});

test('should handle text change', () => {
  const value = 'rent';
  wrapper.find('input').simulate('change', {
    target: { value }
  });
  expect(setTextFilter).toHaveBeenLastCalledWith(value);
});

test('should sort by date', () => {
  const value = 'date';
  wrapper.setProps({
    filters: altFilters
  });
  wrapper.find('select').simulate('change', {
    target: { value }
  });
  expect(sortByDate).toHaveBeenCalled();
});
