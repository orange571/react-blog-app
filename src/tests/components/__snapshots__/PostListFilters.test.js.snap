// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PostListFilters correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__left"
    >
      <div
        className="input-group__item"
      >
        <input
          className="text-input"
          onChange={[Function]}
          placeholder="Search posts"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <select
          className="select"
          onChange={[Function]}
          value="date"
        >
          <option
            value="title"
          >
            Title
          </option>
          <option
            value="date"
          >
            By Date
          </option>
        </select>
      </div>
    </div>
    <div
      className="input-group__item"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Post
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render PostListFilters with alt data correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__left"
    >
      <div
        className="input-group__item"
      >
        <input
          className="text-input"
          onChange={[Function]}
          placeholder="Search posts"
          type="text"
          value="Credit"
        />
      </div>
      <div
        className="input-group__item"
      >
        <select
          className="select"
          onChange={[Function]}
          value="title"
        >
          <option
            value="title"
          >
            Title
          </option>
          <option
            value="date"
          >
            By Date
          </option>
        </select>
      </div>
    </div>
    <div
      className="input-group__item"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Post
      </Link>
    </div>
  </div>
</div>
`;
